### PR TITLE
[feature] Use `local_only` field, deprecate `federated` field

### PR DESCRIPTION
**Pull request by tsmethurst**
# Description

> If this is a code change, please include a summary of what you've coded, and link to the issue(s) it closes/implements.
>
> If this is a documentation change, please briefly describe what you've changed and why.

This pull request updates the status create API and status API model to use the `local_only` field to indicate non-federated statuses. The `local_only` field is already in use by implementations such as GlitchSoc, so this should make things a little easier for client app devs. To avoid breakage, the `federated` field has been deprecated. It still works, but only if `local_only` is not set, and `local_only` takes priority.

Along the way, I also added checks for LocalOnly statuses to the visibility filter. We were already not serving unfederated statuses via the web frontend, but references to local-only posts were still being included in outboxes (not the posts themselves), and I *think* Public local-only posts were included in API responses if `instance-expose-public-timeline` was set to true. Not sure since I don't think anyone was actually using this combination of features.

Closes https://github.com/superseriousbusiness/gotosocial/issues/1035

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have not leveraged AI to create the proposed changes.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [x] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

File: internal/api/model/status.go [hunk 1]
```diff
@@ -209,6 +211,12 @@ type StatusCreateRequest struct {
 	SpoilerText string `form:"spoiler_text" json:"spoiler_text" xml:"spoiler_text"`
 	// Visibility of the posted status.
 	Visibility Visibility `form:"visibility" json:"visibility" xml:"visibility"`
+	// Set to "true" if this status should not be federated, ie. it should be a "local only" status.
+	LocalOnly *bool `form:"local_only"`
+	// ***DEPRECATED***.
+	//
+	// Only used if LocalOnly is not set.
+	DeprecatedFederated *bool `form:"federated"`
```
Comment (NyaaaWhatsUpDoc) [inline internal/api/model/status.go:219]: i think you can probably leave this as `Federated *bool` but if you prefix the comment with `// Deprecated: ` then the linter should always indicate it as deprecated to us developers.

Comment (NyaaaWhatsUpDoc) [reply internal/api/model/status.go:219]: an example if you want to see:

```golang
	// your linter will strike-through
	// your access of r.Cancel, a
	// deprecated field.
	r := http.Request{}
	r.Cancel
```

Comment (NyaaaWhatsUpDoc) [reply internal/api/model/status.go:219]: in action:
deprecate_go_field

Comment (tsmethurst) [reply internal/api/model/status.go:219]: Oh neat!!!

**Pull request merged**